data/crac-file/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCrac.java

```diff
@@ -22,105 +22,197 @@
     }
 
     @Override
-    @JsonProperty("cnecs")
-    public void addCnec(Cnec cnec) {
-        cnecs.add(cnec);
+    public List<State> getStates(Contingency contingency) {
+        return states.stream()
+            .filter(state -> state.getContingency().isPresent() && state.getContingency().get().getId().equals(contingency.getId()))
+            .sorted()
+            .collect(Collectors.toList());
     }
 
-    @JsonProperty("contingency")
     @Override
-    public void addContingency(Contingency contingency) {
-        contingencies.add(contingency);
+    public Set<State> getStates(Instant instant) {
+        return states.stream()
+            .filter(state -> state.getInstant().getId().equals(instant.getId()))
+            .collect(Collectors.toSet());
     }
 
-    @JsonProperty("networkActions")
     @Override
-    public void addNetworkRemedialAction(NetworkAction networkAction) {
-        networkActions.add(networkAction);
+    public State getState(Contingency contingency, Instant instant) {
+        return states.stream()
+            .filter(state -> state.getContingency().isPresent() && state.getInstant().getId().equals(instant.getId()))
+            .filter(state -> state.getContingency().isPresent() && state.getContingency().get().getId().equals(contingency.getId()))
+            .findFirst()
+            .orElse(null);
     }
 
+    /**
+     * Add a state in the Crac object. When adding a state which is made of a Contingency and an Instant,
+     * these Contingency and Instant objects have to be independently present in the Crac as well. So if they
+     * are not, they have to be added as well.
+     * Then the State has to point on the good objects, meaning those which are present independently in the Crac.
+     * So in the end a new State object will be created with references on Contingency and Instant objects
+     * that have been already added to the Crac.
+     *
+     * @param state: state object that can be created from already existing Contingency and Instant object of the Crac
+     *             or not.
+     */
     @Override
-    @JsonProperty("rangeActions")
-    public void addRangeRemedialAction(RangeAction rangeAction) {
-        rangeActions.add(rangeAction);
+    public void addState(State state) {
+        // reference that will point at an object existing in the Crac
+        Optional<Contingency> cracContingency;
+        if (state.getContingency().isPresent()) {
+            Contingency contingency = state.getContingency().get();
+            if (contingencies.stream().noneMatch(contingency::equals)) {
+                addContingency(contingency);
+            }
+            cracContingency = Optional.of(getContingency(contingency.getId()));
+        } else {
+            cracContingency = Optional.empty();
+        }
+
+        // If the two instants are strictly equals no need to add it
```

Comment:
No need to put an -s at the end of "equal" :gb:

Reply:
removed